PROTOCOL.md

```diff
@@ -1348,6 +1348,12 @@ When enabled:
 
 Delta supports defining default expressions for columns on Delta tables. Delta will generate default values for columns when users do not explicitly provide values for them when writing to such tables, or when the user explicitly specifies the `DEFAULT` SQL keyword for any such column.
 
+Semantics for write and read operations:
+- Note that this metadata only applies for write operations, not read operations.
+- SQL DML operations such as INSERT, UPDATE, and MERGE will use the default values. For example: `INSERT INTO t VALUES (42, DEFAULT);`
+- SQL DDL operations such as ALTER TABLE ... ADD COLUMN commands may not specify a default value for any column in the same command that the column is created. For example, this is not supported in Delta Lake: `ALTER TABLE t ADD COLUMN c INT DEFAULT 42;`
```

Comment:
"may not" seems confusing to me? can it or can it not? ideally the protocol should not be tied to SQL operations because non-SQL operations maybe done on Delta table by a non-SQL engine. So it would be good to clarify what exactly cannot be done. As far as I understand .. it seems that a "single operation" that modifies the table schema cannot simultaneously add a new column and also defined its default expression?

Reply:
Sounds good, this is done.